src/file/tokio_compat_file.rs

```diff
@@ -617,36 +641,50 @@ impl AsyncWrite for TokioCompatFile {
 
         let max_write_len = self.max_write_len_impl();
 
-        let (n, bufs, buf) = if let Some(res) = take_io_slices(bufs, max_write_len as usize) {
+        let (n, mut bufs, mut buf) = if let Some(res) = take_io_slices(bufs, max_write_len as usize)
```

Comment:
We can move this to just before the creation of buffers, to avoid use of length_checked:

```suggestion
        let Some(res) = take_io_slices(bufs, max_write_len as usize).map(|res| res.0) else {
```